docs/src/conventions.md

```diff
@@ -0,0 +1,116 @@
+for their configuration.
+
+
+### Exceptions
+
+We recognize that in some cases, the formatting and lints automation may not be applicable to a specific code.
+In those cases, we usually prefer explicitly allowing exceptions by locally disabling the check.
+E.g., use `#[allow]` annotation instead of disabling a link on a crate or project level.
+
+### Copyright notice
+
+All source code files begin with a copyright and license notice. If this is a new file, please add the following notice:
+
+```rust
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+```
+
+When modifying a file from another project, please keep their headers as is and append the following notice after them:
+
+```rust
+// ... existing licensing headers ...
+
+// Modifications Copyright Kani Contributors
+// See GitHub history for details.
+```
+
+We also have automated checks for the copyright notice.
+There are a few file types where this rules doesn't apply.
+You can see that list in the [copyright-exclude](
+https://github.com/model-checking/kani/blob/main/scripts/ci/copyright-exclude) file.
+
+
+## Code for soundness
+
+We are developing Kani to provide assurance that critical Rust components are verifiably free of certain classes of
+security and correctness issues.
+Thus, it is critical that we provide a verification tool is sound.
+For the class of errors that Kani can verify, we should not produce a “No Error” result if there was in fact an
+error in the code being verified, i.e., it has no
+“False Negatives”.
+
+Because of that, we caution on the side of correctness.
+Any incorrect modeling
+that may trigger an unsound analysis that cannot be fixed in the short term should be mitigated.
+Here are a few ways how we do that.
+
+### Compilation errors
+
+Make sure to add user-friendly errors for constructs that we can't handle.
+For example, Kani cannot handle panic unwind strategy, and it will fail compilation if the crate uses this
+configuration.
+
+### Internal compiler errors
+
+Even though this doesn't provide users the best experience, you are encouraged to add checks in the compiler for any
+assumptions you make during development.
+Those check can be on the form of `assert!()` or `unreachable!()`
+statement.
+Please provide a meaningful message to help user understand why something failed, and also help us figure out what
+went wrong.
+
+We don't formally use any specific formal representation of [function contract](https://en.wikipedia.org/wiki/Design_by_contract),
+but whenever possible we do instrument the code with assertions that may represent the function pre- and
+post-conditions to ensure we are modeling the user code correctly.
+
+### Verification errors
+
+In cases where Kani fails to model a certain instruction or local construct that doesn't have a global effect,
+we encode this failure as a verification error.
+I.e., we generate an assertion failure instead of the construct we are modeling using
```

Comment:
We should also clarify that this construct blocks execution.

Reply:
Sure... my brain works with the Rust model where assertions always interrupt the execution. :)